### DISCUSSION TITLE
How to use "-" in URL-Slugs?

### DISCUSSION THREAD

**Discussion by jespermhl** [YOURLS/YOURLS#3804] [Q&A] [answered]
Hello everybody, is there a way to use "-" in URL-Slugs?
If so, how?
Thanks in advance!

**Comment by dgw** [comment, marked as answer]
This feature is [a core plugin bundled with YOURLS](https://github.com/YOURLS/YOURLS/tree/master/user/plugins/hyphens-in-urls).

**Reply by jespermhl** [reply]
Thank you for answering my question so fast! I did not notice that :)

**Comment by logusivam** [comment]
Yes, and you don't even need to write any code — YOURLS **ships an official plugin for exactly this**, it's just not active by default.

1. Log into your admin area and go to **Manage Plugins** (`/admin/plugins.php`).
2. Find **"Allow Hyphens in Short URLs"** (by Ozh) in the list — it's already in your `user/plugins/hyphens-in-urls/` folder.
3. Click **Activate**.

That's it. You can now use hyphens in custom keywords, e.g. `https://sho.rt/hello-world`.

How it works under the hood: the plugin hooks the `get_shorturl_charset` filter and appends `-` to the set of characters YOURLS allows in a keyword. It also deactivates itself just for **auto-generated random keywords**, so those stay hyphen-free while your custom slugs can use hyphens — exactly the behavior you'd want.

If you ever need to allow *additional* characters too (e.g. `@`), you can copy that plugin's folder, rename it, and add the characters in one line — the filter just returns the allowed-character string, so `return $in . '-@';` permits both. But for plain hyphens, activating the bundled plugin is all you need.

Hope this helps — if it answers your question, would you mind clicking **"Mark as answer"**?
